code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -114,6 +114,11 @@
 	if(shoes && (shoes.item_flags & NOSLIP) && istype(shoes, /obj/item/clothing/shoes/magboots))  //magboots + dense_object = no floating
 		return 1
 	return 0
+	
+/mob/living/carbon/human/set_dir(var/new_dir)
+	. = ..()
+	if(species.tail)

```

Comment:
Might as well change this to `if (. && species.tail)`, no reason to recalculate it if dir didn't actually change.